### DISCUSSION TITLE
Multiple queries in one request on an index

### DISCUSSION THREAD

**Discussion by dingusagar** [meilisearch/product#320] [Feedback & Feature Proposal]
Hi, 
I was doing a comparison between the features of meilisearch and typsense for a product search use case. 
One of the features I found interesting in typesense is [federated/multi-search ](https://typesense.org/docs/0.21.0/api/documents.html#federated-multi-search). This is useful when we have multiple search queries as a batch to process. Instead of hitting the search API for each query, all the queries in the batch can be send in a single search API request. 

Wanted to check if meilisearch has any plans to introduce this feature anytime soon?

**Comment by moises-marquez** [comment]
Does the multi-index search feature cover your use case? https://github.com/meilisearch/product/issues/74. It's scheduled for Q1 next year.

**Reply by dingusagar** [reply]
Thanks for the reply. I don't think multi-index search would cover my use-case. Because I just have a single index and I am trying to search multiple queries in one go in a single API call.

**Reply by gmourier** [reply]
Hi 👋

Could you please elaborate on this particular use case for an end-user? What do your documents look like in your index? I guess they don't share the same "scheme". 🤓

In the idea, if we implement a multi-index search, we may also be able to search within the same index with several queries from one single search API call.

Thanks for your feedback!

**Reply by dingusagar** [reply]
Hi, my usecase was to extract product names from end user queries from a chat without doing a NER (since I didn't find any pre-trained NER models for product names that I can use out of the box). the basic idea was to split the query into multiple pieces and search all of them in meilisearch, aggregate the results to find the product names in the query.  for this, I was checking if I can achieve this in a single API call.

**Comment by sandstrom** [comment]
We use ES today, and have some usage of multi-index search.

Basically, we have a global search field where one can type anything, and it will search both the `users` and `companies` indices. Scoring will give some weight to one index. Benefit is that the hits are blended, so scoring is normalized across the two indices. 

It's not critical functionality to us, but certainly useful.

**Comment by gmourier** [comment]
Hi 👋

Adding a multi-index search endpoint has been prioritized to be added during 2022 🏎

I hope to share some tangible news asap!

**Reply by gmourier** [reply]
Hi 👋

We've started a dedicated discussion on the Federated Search / Multi-Index Search subject https://github.com/meilisearch/product/discussions/489 to get more precise details and decide what need to be solved first.

Thanks!

**Comment by gmourier** [comment]
Hey folks ✨

[v1.1](https://github.com/meilisearch/meilisearch/releases/tag/v1.1.0) has been released and features the `/multi-search` endpoint. It allows you to search for multiple indexes in a single HTTP request. ✨

_Note: it's not possible to aggregate results coming from different indexes on a single list natively yet._

📚 https://docs.meilisearch.com/learn/what_is_meilisearch/features.html#multi-index-search
📚 https://docs.meilisearch.com/reference/api/multi_search.html#post-route
